kasa/exceptions.py

```diff
@@ -1,10 +1,15 @@
 """python-kasa exceptions."""
 from enum import IntEnum
+from typing import Optional
 
 
 class SmartDeviceException(Exception):
     """Base exception for device errors."""
 
+    def __init__(self, *args, error_code: Optional["SmartErrorCode"] = None):
+        self.error_code = error_code
+        super().__init__(args)
+
```

Comment:
Please move adding the `error_code` into its own PR ("Add device reported error code to exceptions" or something similar).